packages/terra-consumer-nav/src/components/nav-items/NavItems.jsx

```diff
@@ -42,14 +46,34 @@ class NavItems extends Component {
     this.handleToggle = this.handleToggle.bind(this);
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (this.props !== nextProps) {
+      const currentComponent = this;
+      const currentURL = nextProps.currentURL;
+
+      /* if there is a sub nav which matches the currentURL open the parent nav */
+      currentComponent.props.navItems.forEach((element, i) => {
+        if (element.subItems) {
+          element.subItems.forEach((item) => {
+            if (currentURL === item.url) {
+              currentComponent.setState({
+                openToggle: i,
```

Comment:
There should be only 1 match right? Does it need to keep looping through the rest of them?

Reply:
Good catch! I have updated to find instead of forEach